Fix NU1510 warnings

The build currently generates 4 warnings, which I think are new as of .NET 10:

```text
[WRN] CopyAdditionalDeps: opentelemetry-dotnet-instrumentation\src\OpenTelemetry.AutoInstrumentation.AdditionalDeps\Package.xml : warning NU1510: PackageReference System.Diagnostics.DiagnosticSource will not be pruned. Consider removing this package from your dependencies, as it is likely unnecessary.
[WRN] CopyAdditionalDeps: opentelemetry-dotnet-instrumentation\src\OpenTelemetry.AutoInstrumentation.AdditionalDeps\Package.xml : warning NU1510: PackageReference System.Diagnostics.DiagnosticSource will not be pruned. Consider removing this package from your dependencies, as it is likely unnecessary.
[WRN] CopyAdditionalDeps: opentelemetry-dotnet-instrumentation\src\OpenTelemetry.AutoInstrumentation.AdditionalDeps\Package.xml : warning NU1510: PackageReference System.Diagnostics.DiagnosticSource will not be pruned. Consider removing this package from your dependencies, as it is likely unnecessary.
[WRN] CopyAdditionalDeps: opentelemetry-dotnet-instrumentation\src\OpenTelemetry.AutoInstrumentation.AdditionalDeps\Package.xml : warning NU1510: PackageReference System.Diagnostics.DiagnosticSource will not be pruned. Consider removing this package from your dependencies, as it is likely unnecessary.
```

I believe it's due to this line:

https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/45a3b23a3a7e3d228997b89e922b033791138e82/src/OpenTelemetry.AutoInstrumentation.AdditionalDeps/Directory.Build.props#L5

I'm not sure which of the following three options is correct for this repostory:

1. Remove the reference
2. Make the reference conditional to target framework(s) that do not generate the warning
3. Suppress the `NU1510` warning in `src/OpenTelemetry.AutoInstrumentation.AdditionalDeps/Directory.Build.props`

--- a/src/OpenTelemetry.AutoInstrumentation.AdditionalDeps/Directory.Build.props
+++ b/src/OpenTelemetry.AutoInstrumentation.AdditionalDeps/Directory.Build.props
@@ -1,7 +1,10 @@
 <Project>
+  <PropertyGroup>
+    <NoWarn>$(NoWarn);NU1510</NoWarn>
+  </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.Configuration.Binder" />
     <PackageReference Include="Microsoft.Extensions.Logging.Configuration" />
-    <PackageReference Include="System.Diagnostics.DiagnosticSource" NoWarn="NU1510" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" />
   </ItemGroup>
 </Project>

